identity-iota/src/document/iota_document.rs

```diff
@@ -234,14 +235,18 @@ impl IotaDocument {
     IotaVerificationMethod::check_validity(method)?;
 
     // Ensure the verification method type is supported
-    match method.key_type() {
-      MethodType::Ed25519VerificationKey2018 => {}
-      MethodType::MerkleKeyCollection2021 => return Err(Error::InvalidDocumentSigningMethodType),
+    if !Self::is_signing_method_type(method.key_type()) {
+      return Err(Error::InvalidDocumentSigningMethodType);
     }
 
     Ok(())
   }
 
+  /// Returns whether the given [`MethodType`] can be used to sign document updates.
+  pub fn is_signing_method_type(method_type: MethodType) -> bool {
+    UPDATE_METHOD_TYPES.contains(&method_type)
+  }
```

Comment:
The other usages of `UPDATE_METHOD_TYPES` should probably be replaced with `IotaDocument::is_signing_method_type`, like in `PublishType` and Account `create_identity`.